lib/Common/Memory/WriteBarrierMacros.h

```diff
@@ -17,6 +17,8 @@
 
 #define FieldWithBarrier(type, ...) \
     typename WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type
+#define FieldWithBarrierT(type, ...) \
```

Comment:
Do you mean replace  FieldWithBarrier with  FieldWithBarrierT for legacy ? For declarations within a template we still need  FieldWithBarrier .